### PR TITLE
Update MariaDB Windows libs to v3.2, add Windows startup tests to CI

### PR DESCRIPTION
**Pull request by zach2good**
I affirm:

 - [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
 - [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
 - [x] that I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
 - [x] that I've tested my code and things my code changed since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Review by claywar — APPROVED** [review]
Did a quick sanity with the earlier branch.  Good to go once CI changes are sane.

**Comment by zach2good** [comment]
Still fighting CI, will rebase and sort everything once I have it working

**Comment by zach2good** [comment]
Ok, this is somewhat ready to go now. I'd _really_ appreciate if anyone who sees this can pull it down and test their workflows to make sure I haven't annihilated anything

**Comment by zach2good** [comment]
Reminders for myself:
https://stackoverflow.com/questions/54806802/filtering-critical-fatal-exceptions-for-use-with-addvectoredexceptionhandler
https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-setunhandledexceptionfilter
https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-addvectoredexceptionhandler

TLDR: a vectored exception handler reacts BEFORE any system handlers can go, the unhandled filter goes AFTER
I might be catching `mysql_ping` and terminating the process before we ever get a chance to recover

EDIT: Tested this out, doesn't help with resiliency against `STATUS_ACCESS_VIOLATION`
